snooty/parser.py

```diff
@@ -573,9 +573,11 @@ def dispatch_departure(self, node: tinydocutils.nodes.Node) -> None:
             self.handle_method_option(popped)
 
         elif isinstance(popped, n.Directive) and popped.name == "collapsible":
-            section = popped.get_child_of_type(n.Section)
-            heading = popped.get_child_of_type(n.Heading)
-            if not section and not heading:
+            html5_id = util.make_html5_id(popped.options.get("heading", "")).lower()
+            popped.options["id"] = html5_id
```

Comment:
Will we ever need to ensure uniqueness in collapsible heading IDs like we do with headings in the postprocessor's `HeadingHandler`? I'm thinking of a case where similar collapsible headings are used in method selectors or tabs.

Reply:
done! added logic to ensure uniqueness for heading options in collapsible contents. PTAL and lmk if that works !

Reply:
I think it looks good! Thanks!